Remove WebSocketEnabled & IsNSURLSessionWebSocketEnabled from DeprecatedGlobalSettings

nt1m opened the pull request:
#### c3b5f312a71d8598ff1727d406a0782074fbd35f
<pre>
Remove WebSocketEnabled &amp; IsNSURLSessionWebSocketEnabled from DeprecatedGlobalSettings
<a href="https://bugs.webkit.org/show_bug.cgi?id=250244">https://bugs.webkit.org/show_bug.cgi?id=250244</a>
rdar://103973264

Reviewed by Brent Fulgham and Tim Horton.

Use settingsValues and [EnabledBySetting].

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/websockets/ThreadableWebSocketChannel.cpp:
(WebCore::ThreadableWebSocketChannel::create):
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::send):
* Source/WebCore/Modules/websockets/WebSocket.idl:
* Source/WebCore/page/DeprecatedGlobalSettings.cpp:
(WebCore::DeprecatedGlobalSettings::DeprecatedGlobalSettings):
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setWebSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::webSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::isNSURLSessionWebSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::setIsNSURLSessionWebSocketEnabled): Deleted.

scriptExecutionContext() should always exist given it is passed with the constructor as a reference.

Canonical link: <a href="https://commits.webkit.org/258602@main">https://commits.webkit.org/258602@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb4d4e2dc846b375d216b76d663070254c11128e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102478 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11612 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35512 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111746 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/171959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106447 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2513 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109465 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108258 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9624 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/92895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/24379 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/92729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/5074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/25798 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/89047 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/2753 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5231 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29596 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11250 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45295 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/91972 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5908 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/6967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20600 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -6883,6 +6884,17 @@ WebShareFileAPIEnabled:
     WebCore:
       default: false
 
+WebSocketsEnabled:
```
I'm not sure we really need this setting flag. Can modern websites really work without WebSockets?

brentfulgham commented on Source/WebCore/Modules/websockets/WebSocket.idl:
```diff
@@ -34,7 +34,7 @@
     ActiveDOMObject,
     Exposed=(Window,Worker),
-    EnabledByDeprecatedGlobalSetting=WebSocketEnabled
+    EnabledBySetting=WebSocketEnabled
```
Where was this coming from? The 'DeprecatedGlobalSetting' was `IsNSURLSessionWebSocketEnabled`, so how did this ever get set before?

nt1m replied:
There was another DeprecatedGlobalSetting, see my changes in DeprecatedGlobalSettings.h . Might be worth removing the flag entirely, but I'd rather do that in a separate bug

nt1m replied:
Turns out we had a preference for WebSocket so we could disable it on WatchOS!

brentfulgham reviewed (approved):
Looks like this might have been broken-ish for a while...

nt1m committed:
Remove WebSocketEnabled & IsNSURLSessionWebSocketEnabled from DeprecatedGlobalSettings
https://bugs.webkit.org/show_bug.cgi?id=250244
rdar://103973264

Reviewed by Brent Fulgham and Tim Horton.

Use settingsValues and [EnabledBySetting].

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/websockets/ThreadableWebSocketChannel.cpp:
(WebCore::ThreadableWebSocketChannel::create):
* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::send):
* Source/WebCore/Modules/websockets/WebSocket.idl:
* Source/WebCore/page/DeprecatedGlobalSettings.cpp:
(WebCore::DeprecatedGlobalSettings::DeprecatedGlobalSettings):
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setWebSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::webSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::isNSURLSessionWebSocketEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::setIsNSURLSessionWebSocketEnabled): Deleted.

scriptExecutionContext() should always exist given it is passed with the constructor as a reference.

Canonical link: https://commits.webkit.org/258602@main
